syntax = "proto3";
package fusion.v1;

message Detection {
  string id = 1;
  int64  ts_unix_ms = 2;
  double lat = 3;
  double lon = 4;
  double confidence = 5;
  string source = 6;
  string schema_version = 7;  // prevents ambiguity and helps with scalability
}

message SubscribeRequest {
  bool fused_only = 1;
}

service FusionStream {
  rpc Subscribe(SubscribeRequest) returns (stream Detection);
}
